targets/TARGET_STM/hal_tick_overrides.c

```diff
@@ -27,11 +27,10 @@ extern int mbed_sdk_inited;
 void init_16bit_timer(void);
 void init_32bit_timer(void);
 
-#if TIM_MST_BIT_WIDTH == 16
 // Variables also reset in us_ticker_init()
+uint32_t total_ticks = 0;
```

Comment:
I think you can make these `static` - see other comment.

Just to potentially save 2 bytes of RAM, I think I'd have the type  of `prev_time` conditional on `TIM_MST_BIT_WIDTH <= 16`